src/renderer/pages/Operations/common/utils.ts

```diff
@@ -218,3 +233,37 @@ export const getSignatoryName = (
 
   return toAddress(signatoryId, { chunk: 5, prefix: addressPrefix });
 };
+
+export const getSignatoryAccounts = (
+  accounts: Account[],
+  wallets: Wallet[],
+  events: MultisigEvent[],
+  signatories: Signatory[],
+  chainId: ChainId,
+) => {
+  const walletsMap = new Map(wallets.map((wallet) => [wallet.id, wallet]));
+
+  return signatories.reduce((acc: Account[], signatory) => {
+    const filteredAccounts = accounts.filter(
+      (a) => a.accountId === signatory.accountId && !events.some((e) => e.accountId === a.accountId),
+    );
+
+    const signatoryAccount = filteredAccounts.find((a) => {
+      const isChainMatch = accountUtils.isChainIdMatch(a, chainId);
+      const wallet = walletsMap.get(a.walletId);
+
+      return isChainMatch && walletUtils.isWalidSignatory(wallet);
```

Comment:
Don't forget to fix `isWalidSignatory` 😉

Reply:
Let's say I have 1 Legacy Multisig account with 3 signatories:
- **my** Signle_shard_parity_signer (root)
- **my** Multishard_parity_signer (root)
- friend's account
According to `isValidSignatory` Single_shard is the only one I will see
Seems to me that I should be able to choose from 2 signatories (single_shard & multishard), am I right?
@pgolovkin Please, correct me if I'm wrong from the business perspective

Reply:
This case is handled in `getSignatoryAccount` where `isValidSignatory ` is used. I think `isValidSignatory` should represetn what type of wallet can we use as signatory in current state and legacy case should be handled outside of this method